src/core/stdcpp/typeinfo.d

```diff
@@ -71,11 +71,11 @@ else version (CppRuntime_Microsoft)
     class type_info
     {
         //virtual ~this();
-        void dtor() { }     // reserve slot in vtbl[]
+        void dtor() nothrow { }     // reserve slot in vtbl[]
         //bool operator==(const type_info rhs) const;
         //bool operator!=(const type_info rhs) const;
-        final bool before(const type_info rhs) const;
-        final const(char)* name(__type_info_node* p = &__type_info_root_node) const;
+        final bool before(const type_info rhs) const nothrow;
+        final const(char)* name(__type_info_node* p = &__type_info_root_node) const nothrow;
```

Comment:
That's actually the wrong signature for VS 2017 (which takes no params), but I guess DMD has to maintain VS 2010 compatibility or whatever.

Reply:
We have `-extern-std=`, so we can version the C++ code according to the C++ version it binds to.